src/com/google/javascript/jscomp/ConvertChunksToESModules.java

```diff
@@ -0,0 +1,297 @@
+        } else {
+          Node firstInputAstRoot = firstInput.getAstRoot(compiler);
+          FeatureSet firstInputScriptFeatures = NodeUtil.getFeatureSetOfScript(firstInputAstRoot);
+          FeatureSet combinedFeatureSet =
+              firstInputScriptFeatures.union(NodeUtil.getFeatureSetOfScript(astRoot));
+          astRoot.putProp(Node.FEATURE_SET, combinedFeatureSet);
+          Node moduleBody = firstInputAstRoot.getFirstChild();
+          checkState(moduleBody != null && moduleBody.isModuleBody());
+          moduleBody.addChildrenToBack(astRoot.removeChildren());
+          compiler.reportChangeToEnclosingScope(firstInputAstRoot);
+          compiler.reportChangeToChangeScope(astRoot);
+        }
+      }
+    }
+  }
+
+  /**
+   * Add export statements to chunks
+   */
+  private void AddExportStatements() {
+    for (Map.Entry<JSModule, Set<String>> jsModuleExports : crossChunkExports.entrySet()) {
+      CompilerInput firstInput = jsModuleExports.getKey().getInput(0);
+      Node moduleBody = firstInput.getAstRoot(compiler).getFirstChild();
+      checkState(moduleBody != null && moduleBody.isModuleBody());
+      Node exportSpecs = new Node(Token.EXPORT_SPECS);
+      for (String name : jsModuleExports.getValue()) {
+        Node exportSpec = new Node(Token.EXPORT_SPEC);
+        exportSpec.addChildToFront(IR.name(name));
+        exportSpec.addChildToFront(IR.name(name));
+        exportSpec.putIntProp(Node.IS_SHORTHAND_PROPERTY, 1);
+        exportSpecs.addChildToBack(exportSpec);
+      }
+      Node export = IR.export(exportSpecs).useSourceInfoFromForTree(moduleBody);
+      moduleBody.addChildToBack(export);
+      compiler.reportChangeToEnclosingScope(moduleBody);
+    }
+  }
+
+  private static String GetChunkName(JSModule chunk) {
+    return chunk.getName() + ".js";
+  }
+
+  /**
+   * Add import statements to chunks
+   */
+  private void AddImportStatements() {
+    for (Map.Entry<JSModule, Map<JSModule, Set<String>>> chunkImportsEntry :
+        crossChunkImports.entrySet()) {
+      ArrayList<Node> importStatements = new ArrayList<>();
+      JSModule importingChunk = chunkImportsEntry.getKey();
+      CompilerInput firstInput = importingChunk.getInput(0);
+      Node moduleBody = firstInput.getAstRoot(compiler).getFirstChild();
+      checkState(moduleBody != null && moduleBody.isModuleBody());
+
+      // For each distinct chunk where a referenced symbol is defined, create an import statement
+      // referencing the names.
+      for (Map.Entry<JSModule, Set<String>> importsByChunk : chunkImportsEntry.getValue().entrySet()) {
+        Node importSpecs = new Node(Token.IMPORT_SPECS);
+        for (String name : importsByChunk.getValue()) {
+          Node importSpec = new Node(Token.IMPORT_SPEC);
+          importSpec.addChildToFront(IR.name(name));
+          importSpec.addChildToFront(IR.name(name));
+          importSpec.putIntProp(Node.IS_SHORTHAND_PROPERTY, 1);
```

Comment:
use `importSpec.setShorthandProperty(true)` instead of `putIntProp(1)`
Also other places `IS_SHORTHAND_PROPERTY` is used.

Reply:
Didn't know about that one - much nicer